### PR TITLE
Define and activate AWS 9.3.4 release

### PR DESCRIPTION
**Pull request by sslavic**
Towards giantswarm/giantswarm#11530

### CODE REVIEW

File: aws/v9.3.4/release.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - name: cert-operator
+      version: 0.1.0
+    - name: cluster-operator
+      version: 0.23.10
+    - name: kubernetes
+      version: 1.16.9
+    - name: containerlinux
+      version: 2345.3.1
+    - name: calico
+      version: 3.10.1
```
Comment (cokiengchiara) [inline aws/v9.3.4/release.yaml:47]: Did Firecracker make the required changes in aws-operator here? For including newer Calico https://github.com/giantswarm/giantswarm/issues/11307

**Comment by cokiengchiara** [comment]
@sslavic , external-dns is similar to cert-manager in that it is only AWS 10+ is that correct?

File: aws/v9.3.4/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# :zap: Giant Swarm Release v9.3.4 for AWS :zap:
+
+This release updates managed apps to the latest releases.
```
Comment (cokiengchiara) [inline aws/v9.3.4/README.md:3]: @sslavic release notes look fine to me from Halo perspective. 

What about the other components? Nothing notable? cc @MartaD for Ludacris components

**Comment by MartaD** [comment]
Looks like changing calico is working: https://github.com/giantswarm/giantswarm/issues/11307#issuecomment-643236588 but would have to be tested. In case it works, release notes can be copied: 

### calico v3.10.4

 - Fixes IPv6 rogue router advertisement vulnerability
  [CVE-2020-13597](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13597).

 Complete details for changes since v3.10.1 please check the upstream release notes at
https://docs.projectcalico.org/archive/v3.10/release-notes/

**Comment by cokiengchiara** [comment]
I checked AWS 9.3.3 for upgrade notes that needs to persist. There is none.

File: aws/v9.3.4/release.yaml (hunk 1 above)
Comment (sslavic) [reply aws/v9.3.4/release.yaml:47]: @giantswarm/team-firecracker is there aws-operator legacy release with calico upgrade?

File: aws/v9.3.4/README.md (hunk 2 above)
Comment (sslavic) [reply aws/v9.3.4/README.md:3]: Yeah, nothing notable, no special highlights.

Comment (MartaD) [reply aws/v9.3.4/README.md:3]: I’ve checked with the team and we would like to ask to include fixes for 3 CVEs:
- k8s 1.16.11 fresh from today, not communicated yet
- Flatcar 2512.2.1 - newest stable
- calico 3.10.4 - this aws-operator version is from before CNI so it’s not hard coded anywhere

File: aws/v9.3.4/release.yaml (hunk 1 above)
Comment (tfussell) [reply aws/v9.3.4/release.yaml]: The specified aws-operator, v5.7.1, uses k8scloudconfig v6.1.1 which accepts Calico version from the Release CR so this should work. See https://github.com/giantswarm/k8scloudconfig/blob/v6.1.1/files/k8s-resource/calico-all.yaml#L380

Comment (sslavic) [reply aws/v9.3.4/release.yaml]: Thanks @tfussell, will give it another try. Last time I was testing it, it didn't work well with cluster upgrades - calico was remaining on same old version from previous platform release.

Comment (tfussell) [reply aws/v9.3.4/release.yaml]: Yes, it should work. Let me know if you have a cluster where it doesn't work, and I'll take a look.

**Comment by sslavic** [comment]
Tested on giraffe (aws, china)
- 9.3.4 cluster creation
- upgrade from 9.3.3 to 9.3.4

File: aws/v9.3.4/release.yaml (hunk 1 above)
Comment (sslavic) [reply aws/v9.3.4/release.yaml]: It worked this time. Included calico 3.10.4 upgrade in the 9.3.4 release and release notes.

**Comment by sslavic** [comment]
@cokiengchiara @MartaD PTAL at release notes and confirm release announcement is ready, i.e. you're good with shipping this release.

File: aws/v9.3.4/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
```
Comment (cokiengchiara) [inline aws/v9.3.4/README.md:3]:
```suggestion
This release updates managed apps to the latest releases. It also includes security fixes for 3 CVEs.
```

Comment (MartaD) [reply aws/v9.3.4/README.md:3]: I've changed it in the [other PR](https://github.com/giantswarm/releases/pull/344/files#diff-46bc0c7a2ace3de49a04a338c907e701 just now. BTW should I merge that other PR here to have less confusion? @sslavic

File: aws/v9.3.4/release.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    version: 1.2.0
+  components:
+  - name: app-operator
+    version: 1.0.0
+  - name: aws-operator
+    version: 5.7.1
+  - name: cert-operator
+    version: 0.1.0
+  - name: cluster-operator
+    version: 0.23.10
```
Comment (sslavic) [inline aws/v9.3.4/release.yaml:40]:
```suggestion
    version: 0.23.12
```
TODO once it's released

File: aws/v9.3.4/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+- Moved to App deployment.
+
+### cluster-autoscaler v1.16.5 ([Giant Swarm app v1.16.0](https://github.com/giantswarm/cluster-autoscaler-app/blob/master/CHANGELOG.md#v1160-2020-05-26))
+
+- Upgraded upstream cluster-autoscaler from v1.16.2 to v1.16.5 - [changelog](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.16.5).
+- Set `scan-interval` to 30 seconds (from 10 seconds) to save resources.
+- Set `scale-down-unneeded-time` to 5 minutes (from the default of 10 minutes) to release unneeded nodes earlier.
+- Lower `scaleDownUtilizationThreshold` to 0.5.
+
+### cluster-operator [v0.23.10](https://github.com/giantswarm/cluster-operator/releases/tag/v0.23.10)
```
Comment (sslavic) [inline aws/v9.3.4/README.md:29]:
```suggestion
### cluster-operator [v0.23.12](https://github.com/giantswarm/cluster-operator/releases/tag/v0.23.12)
```

TODO once it's released

**Comment by sslavic** [comment]
9.3.4 retest failed on giraffe (aws china), it seems flatcar 2512.2.1 image has not been distributed yet to china.

@mazzy89 can you please help with provisioning the image? Do we depend on https://github.com/giantswarm/giantswarm/issues/11297 ? I see https://github.com/giantswarm/flatcar-china builds were failing and canceled https://circleci.com/gh/giantswarm/flatcar-china


Anyway, once image gets provisioned we can retest and if all OK, ship the release. Until then, this is blocked.

**Comment by sslavic** [comment]
Flatcar upgrade has been removed from the release since it requires aws-operator legacy to be prepared and released with change like done for aws-operator master to support the new flatcar version https://github.com/giantswarm/aws-operator/pull/2543/files

Without that support built-in, with just declared flatcar version updated in platform release definition, platform release tests were failing, tenant cluster was not coming up, aws-operator 5.7.1 was reporting errors like:

```
{"caller":"github.com/giantswarm/operatorkit/resource/wrapper/retryresource/basic_resource.go:59","controller":"aws-operator","event":"update","level":"warning","loop":"141","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5ksu9","resource":"tccp","stack":{"annotation":"no image id for version '2512.2.1'","kind":"invalidConfigError","stack":[{"file":"/go/src/github.com/giantswarm/aws-operator/service/controller/key/key.go","line":728},{"file":"/go/src/github.com/giantswarm/aws-operator/service/controller/resource/tccp/create.go","line":289},{"file":"/go/src/github.com/giantswarm/aws-operator/service/controller/resource/tccp/create.go","line":168},{"file":"/go/src/github.com/giantswarm/aws-operator/service/controller/resource/tccp/create.go","line":93},{"file":"/go/pkg/mod/github.com/giantswarm/operatorkit@v0.2.0/resource/wrapper/retryresource/basic_resource.go","line":52}]},"time":"2020-07-11T08:01:25.83048+00:00","version":"275715566"}
```

**Comment by sslavic** [comment]
@cokiengchiara I retested aws 9.3.4 on giraffe (aws china), after removing flatcar upgrade it works again:
- 9.3.4 cluster creation
- upgrade from 9.3.3

Shall we ship it? Announcement ready or it needs updating due to flatcar upgrade being removed?

File: aws/v9.3.4/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# :zap: Giant Swarm Release v9.3.4 for AWS :zap:
+
+This release updates managed apps to the latest releases and includes security fixes for 3 CVEs:
```
Comment (cokiengchiara) [inline aws/v9.3.4/README.md:3]:
```suggestion
This release makes security fixes for 3 CVEs:
```

File: aws/v9.3.4/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+- IPv6 rogue router advertisement vulnerability [CVE-2020-13597](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13597)
+- Intel Microcode vulnerabilities [CVE-2020-0543](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-0543)
+- [CVE-2020-8558](https://github.com/kubernetes/kubernetes/issues/92315) which allows for neighboring hosts to bypass localhost boundary
+
```
Comment (cokiengchiara) [inline aws/v9.3.4/README.md:7]:
```suggestion

It also updates managed apps to the latest releases.

```

I saw @sslavic feedback on sig-product and think it makes sense. wdyt @sslavic ?

File: aws/v9.3.4/README.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
+# :zap: Giant Swarm Release v9.3.4 for AWS :zap:
+
+This release updates managed apps to the latest releases and includes security fixes for 3 CVEs:
+- IPv6 rogue router advertisement vulnerability [CVE-2020-13597](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13597)
+- Intel Microcode vulnerabilities [CVE-2020-0543](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-0543)
```
Comment (stone-z) [inline aws/v9.3.4/README.md:5]: It no longer includes this

File: aws/v9.3.4/README.md (hunk 2 above)
Comment (sslavic) [reply aws/v9.3.4/README.md]: k8s + calico upgrade was included, flatcar requires aws-operator legacy update which is not available yet so will be shipped separately.

**Comment by cokiengchiara** [comment]
> Announcement ready or it needs updating due to flatcar upgrade being removed?

Am I blind? I don't see any mention of flatcar? Did you remove already? haha

File: aws/v9.3.4/README.md (hunk 3 above)
Comment (sslavic) [reply aws/v9.3.4/README.md]: Other PR got merged.

File: aws/v9.3.4/README.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# :zap: Giant Swarm Release v9.3.4 for AWS :zap:
+
+This release makes security fixes for 2 CVEs:
+- IPv6 rogue router advertisement vulnerability [CVE-2020-13597](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13597)
+- Intel Microcode vulnerabilities [CVE-2020-0543](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-0543)
```
Comment (cokiengchiara) [inline aws/v9.3.4/README.md:5]:
```suggestion
```

@stone-z says this is not there. Is this the flatcar one?

File: aws/v9.3.4/README.md (hunk 8 above)
Comment (sslavic) [reply aws/v9.3.4/README.md:5]: @stone-z yes. please include suggestion to remove these

Comment (sslavic) [reply aws/v9.3.4/README.md:5]: Chiara did it https://github.com/giantswarm/releases/pull/337#discussion_r453740085

File: aws/v9.3.4/README.md (hunk 9 above)
Comment (stone-z) [reply aws/v9.3.4/README.md]: Correct, CVE-2020-0543 was the actual vulnerability, the mitigation is from Flatcar

**Comment by sslavic** [comment]
After adding coredns app 1.2.0 to the aws 9.3.4 release, I retested the release on giraffe, 9.3.4 cluster creation works.
Shipping

**Pull request merged**